libcudacxx/include/cuda/__complex/complex.h

```diff
@@ -0,0 +1,237 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of libcu++, the C++ Standard Library for your entire system,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _CUDA___COMPLEX_COMPLEX_H
+#define _CUDA___COMPLEX_COMPLEX_H
+
+#include <cuda/std/detail/__config>
+
+#if defined(_CCCL_IMPLICIT_SYSTEM_HEADER_GCC)
+#  pragma GCC system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_CLANG)
+#  pragma clang system_header
+#elif defined(_CCCL_IMPLICIT_SYSTEM_HEADER_MSVC)
+#  pragma system_header
+#endif // no system header
+
+#include <cuda/__complex/get_real_imag.h>
+#include <cuda/__complex/traits.h>
+#include <cuda/__fwd/complex.h>
+#include <cuda/std/__complex/complex.h>
+#include <cuda/std/__concepts/concept_macros.h>
+#include <cuda/std/__cstddef/types.h>
+#include <cuda/std/__floating_point/conversion_rank_order.h>
+#include <cuda/std/__floating_point/traits.h>
+#include <cuda/std/__type_traits/is_same.h>
+
+#include <cuda/std/__cccl/prologue.h>
+
+_CCCL_BEGIN_NAMESPACE_CUDA
+
+//! @brief Class representing a complex number.
+//!
+//! @tparam _Tp The type of the real and imaginary parts. Must be a NumericType.
+//!
+//! @note Compared to std::complex, this type is trivially default constructible and is aligned to 2 * sizeof(_Tp).
+template <class _Tp>
+class _CCCL_TYPE_VISIBILITY_DEFAULT _CCCL_ALIGNAS(2 * sizeof(_Tp)) complex
+{
+  static_assert(!::cuda::std::__is_ext_nv_fp_v<_Tp>, "Extended NVIDIA floating point types are not supported yet.");
+
+  _Tp __re_; //!< The real part of the complex number.
+  _Tp __im_; //!< The imaginary part of the complex number.
+
+public:
+  using value_type = _Tp; //!< The type of the values.
+
+  //! @brief Trivial default constructor.
+  _CCCL_HIDE_FROM_ABI constexpr complex() noexcept = default;
+
+  //! @brief Constructs a complex number with the given real and imaginary parts.
+  //!
+  //! @param __re The real part of the complex number.
+  //! @param __im The imaginary part of the complex number, defaults to 0.
+  _CCCL_API constexpr complex(const _Tp& __re, const _Tp& __im = _Tp{}) noexcept
+      : __re_{__re}
+      , __im_{__im}
+  {}
+
+  //! @brief Defaulted copy constructor.
+  _CCCL_HIDE_FROM_ABI constexpr complex(const complex&) noexcept = default;
+
+  //! @brief Constructs a complex number from another complex number of a different type.
```

Comment:
Could we simplify the whole thing as
```suggestion
  //! @brief Constructs a complex number from another complex number @param __other of a different type.
```

Reply:
I don't think this is correct, it produces very ugly doxygen docs